fluss-flink/fluss-flink-common/src/main/java/com/alibaba/fluss/flink/sink/FlussSinkBuilder.java

```diff
@@ -0,0 +1,221 @@
+import com.alibaba.fluss.metadata.DataLakeFormat;
+import com.alibaba.fluss.metadata.TableInfo;
+import com.alibaba.fluss.metadata.TablePath;
+
+import org.apache.flink.table.types.logical.RowType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+import static com.alibaba.fluss.utils.Preconditions.checkArgument;
+import static com.alibaba.fluss.utils.Preconditions.checkNotNull;
+
+/**
+ * Builder for creating and configuring Fluss sink connectors for Apache Flink.
+ *
+ * <p>The builder supports automatic schema inference from POJO classes using reflection and
+ * provides options for customizing data conversion logic through custom converters.
+ *
+ * <p>Example usage:
+ *
+ * <pre>{@code
+ * FlinkSink<Order> sink = new FlussSinkBuilder<Order>()
+ *          .setBootstrapServers(bootstrapServers)
+ *          .setTable(tableName)
+ *          .setDatabase(databaseName)
+ *          .setRowType(orderRowType)
+ *          .setSerializationSchema(new OrderSerializationSchema())
+ *          .build())
+ * }</pre>
+ *
+ * @param <InputT>> The input type of records to be written to Fluss
+ */
+public class FlussSinkBuilder<InputT> {
+    private static final Logger LOG = LoggerFactory.getLogger(FlussSinkBuilder.class);
+
+    private String bootstrapServers;
+    private String database;
+    private String tableName;
+    private RowType tableRowType;
+    private boolean ignoreDelete;
+    private int[] partialUpdateColumns;
+    //    private boolean isUpsert;
+    private final Map<String, String> configOptions = new HashMap<>();
+    private FlussSerializationSchema<InputT> serializationSchema;
+    private boolean shuffleByBucketId = true;
+    private @Nullable DataLakeFormat lakeFormat;
+
+    /** Set the bootstrap server for the sink. */
+    public FlussSinkBuilder<InputT> setBootstrapServers(String bootstrapServers) {
+        this.bootstrapServers = bootstrapServers;
+        return this;
+    }
+
+    /** Set the database for the sink. */
+    public FlussSinkBuilder<InputT> setDatabase(String database) {
+        this.database = database;
+        return this;
+    }
+
+    /** Set the table name for the sink. */
+    public FlussSinkBuilder<InputT> setTable(String table) {
+        this.tableName = table;
+        return this;
+    }
+
+    /** Set the row type for the sink. */
+    public FlussSinkBuilder<InputT> setRowType(RowType rowType) {
+        this.tableRowType = rowType;
+        return this;
+    }
+
+    /** Set whether to ignore delete operations. */
+    public FlussSinkBuilder<InputT> setIgnoreDelete(boolean ignoreDelete) {
+        this.ignoreDelete = ignoreDelete;
+        return this;
+    }
+
+    /** Set target column indexes. */
+    public FlussSinkBuilder<InputT> setPartialUpdateColumns(int[] partialUpdateColumns) {
```

Comment:
I removed this method because the builder doesn't currently support partial updates. We can reintroduce it once partial update support to the DataStream API in the future. Since it's a public API, making changes after release will be difficult, so it's better to wait until the feature is ready. Additionally, I think using field names instead of field indexes would be more user-friendly and less error-prone.